src/sth_database.js

```diff
@@ -5,7 +5,109 @@
 
   var mongoose = require('mongoose');
 
-  var sthConfig, sthLogger, sthHelper, connectionURL, eventSchema, aggregatedSchema;
+  var sthConfig, sthLogger, sthHelper, connectionURL, connection, eventSchema, aggregatedSchema;
+
+  /**
+   * Declares the Mongoose schemas.
+   */
+  function defineSchemas() {
+    switch (sthConfig.DATA_MODEL) {
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_SERVICE_PATH:
+        eventSchema = mongoose.Schema({
+          timestamp: Date,
+          entityId: String,
+          entityType: String,
+          attributeName: String,
+          attributeType: String,
+          attributeValue: Number
+        });
+        break;
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_ENTITY:
+        eventSchema = mongoose.Schema({
+          timestamp: Date,
+          attributeName: String,
+          attributeType: String,
+          attributeValue: Number
+        });
+        break;
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_ATTRIBUTE:
+        eventSchema = mongoose.Schema({
+          timestamp: Date,
+          attributeType: String,
```

Comment:
The attributeType field has been brought back to life :p as a consequence of the discussion we've had with @fgalan where he told us that the attribute type of a concrete attribute name could change over time and consequently cannot be used to generate the name of the collections but stored in the docs themselves ;)